web/src/main/kotlin/com/example/ajaxproject/bpp/NotificationAnnotationBeanPostProcessor.kt

```diff
@@ -5,88 +5,110 @@ import com.example.ajaxproject.dto.request.EmailDTO
                 serviceBeans[beanName] = beanClass
+                beanAnnotatedMethods[beanName] = annotatedMethods
             }
         }
-        return super.postProcessBeforeInitialization(bean, beanName)
+        return bean
     }
 
     override fun postProcessAfterInitialization(bean: Any, beanName: String): Any? {
         return if (serviceBeans.containsKey(beanName)) {
             Proxy.newProxyInstance(
                 bean::class.java.classLoader,
                 bean::class.java.interfaces,
-                NotificationInvocationHandler(bean, beanFactory)
+                NotificationInvocationHandler(bean, beanFactory, beanAnnotatedMethods[beanName] ?: emptyList())
             )
         } else {
             bean
         }
     }
+
+    private fun findAnnotatedMethods(beanClass: Class<*>): List<Method> {
+        return beanClass.methods.filter { it.isAnnotationPresent(Notification::class.java) }
+    }
 }
+
+
 @Suppress("SpreadOperator")
 class NotificationInvocationHandler(
     private val bean: Any,
-    private val beanFactory: BeanFactory
+    private val beanFactory: BeanFactory,
+    private val annotatedMethods: List<Method>
 ) : InvocationHandler {
     override fun invoke(proxy: Any, method: Method, args: Array<out Any>?): Any? {
         val result = method.invoke(bean, *args.orEmpty())
 
-        if (args?.any { it is Identifiable } == true) {
-            createNotification(args.find { it is Identifiable } as Identifiable)
+        for (annotatedMethod in annotatedMethods) {
+            if (method.name == annotatedMethod.name) {
+                val identifiableArg = args?.find { it is Identifiable } as? Identifiable
+                if (identifiableArg != null) {
+                    createNotification(identifiableArg)
+                }
+            }
         }
         return result
     }
 
     private fun createNotification(identifiable: Identifiable) {
 
         val groupChatRoomRepository = beanFactory.getBean(GroupChatRoomRepository::class.java)
-        val chat = groupChatRoomRepository.findChatRoom(identifiable.chatId)
-        val chatName = chat.chatName
-        val userList = chat.chatMembers
-
         val emailSenderService = beanFactory.getBean(EmailSenderService::class.java)
+        beanFactory.getBean("sendEmailThreadPool", ExecutorService::class.java)
 
-        val executorService = beanFactory.getBean("sendEmailThreadPool", ExecutorService::class.java)
+        val chatMono = groupChatRoomRepository.findChatRoom(identifiable.chatId)
 
-        userList.asSequence()
-            .map { user -> buildEmail(user.email, chatName) }
-            .forEach { email ->
-                executorService.submit {
-                    emailSenderService.send(email)
+        val userListFlux = chatMono.flatMapMany { chat ->
+            Flux.fromIterable(chat.chatMembers)
+        }
+
+        userListFlux
+            .flatMap { user ->
+                chatMono.map { chat ->
```

Comment:
Try to execute the following code:
```kotlin
    val checkFlag = AtomicBoolean(false)
    val mono1 = Mono.fromSupplier { "mono supplier" }.doOnNext {
        if (checkFlag.getAndSet(true)) {
            throw Error("Mono invoked twice")
        }
        println("mono 1 supplier")
    }

    val flux: Flux<Int> = listOf(1, 2, 3).toFlux()

    flux
        .flatMap { fluxVal ->
            mono1.map { monoVal -> "flux=$fluxVal, mono=$monoVal" }
        }
        .doOnNext { println(it) }
        .subscribe()
```

This code is analogue to your `userListFlux` + `chatMono`

Reply:
Updated all my reactor code in bpp